huaweicloud/services/modelarts/resource_huaweicloud_modelarts_model.go

```diff
@@ -0,0 +1,726 @@
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Model runtime environment.`,
+			},
+			"metrics": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Model precision.`,
+			},
+			"source_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Model source type`,
+			},
+			"dependencies": {
+				Type:        schema.TypeList,
+				Elem:        modelartsModelDependencySchema(),
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Package required for inference code and model.`,
+			},
+			"workspace_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Workspace ID, which defaults to 0.`,
+			},
+			"model_algorithm": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Model algorithm.`,
+			},
+			"install_type": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Deployment type. Only lowercase letters are supported.`,
+			},
+			"schema_doc": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Download address of the model schema file.`,
+			},
+			"image_address": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Image path generated after model packaging.`,
+			},
+			"model_size": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Model size, in bytes.`,
+			},
+			"model_status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Model status.`,
+			},
+			"model_source": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Model source.`,
+			},
+			"tunable": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: `Whether a model can be tuned.`,
+			},
+			"market_flag": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: `Whether a model is subscribed from AI Gallery.`,
+			},
+			"publishable_flag": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: `Whether a model is subscribed from AI Gallery.`,
+			},
+			"created_at": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Time when a model is created, in milliseconds calculated from 1970.1.1 0:0:0 UTC.`,
+			},
+		},
+	}
+}
+
+func modelartsModelModelDocsSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"doc_url": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `HTTP(S) link of the document.`,
```

Comment:
add forcenew here, in list, if the parameter of element changed, that will not cause the whole list change

Reply:
added